### ISSUE TITLE
Allow running on Linux without root

### ISSUE THREAD

**Issue by mcginty** [tonarino/innernet#41]
Linux users should be able to create an `innernet` user that has permissions to `/etc/innernet` and `/var/lib/innernet`, and the binaries should be runnable after giving it `CAP_NET_ADMIN` permission.

As far as I know, no such equivalent exists on macOS.

**Comment by mcginty** [comment]
Yeah, I really wish Linux had a `CAP_NET_TUN` that was more constricted, but no such capability exists. If anyone was interested in creating SELinux profiles for innernet, though, I think that would be a great addition to the official documentation/packages.

**Comment by mk-fg** [comment]
I've made this one to run it with AppArmor here:
```
#include <tunables/global>

profile innd /usr/bin/innernet-server {
  /usr/bin/innernet-server rm,
  /etc/innernet-server/ rw,
  /etc/innernet-server/* rw,
  /var/lib/innernet-server/ rw,
  /var/lib/innernet-server/* rwlk,

  /usr/lib/** rm,
  /etc/ld.so.* r,
  /dev/urandom r,
  /dev/null rw,
  owner @{PROC}/@{pid}/{maps,cgroup} r,

  network inet,
  capability net_admin,
  signal (receive) peer=unconfined,
}
```
SELinux one can probably have similar set of rules, just in its own syntax.

**Comment by mk-fg** [comment]
> If anyone was interested in creating SELinux profiles for innernet, though, I think that would be a great addition to the official documentation/packages.

I've made this AppArmor profile to run innernet-server locally:
```
abi <abi/3.0>,
#include <tunables/global>

profile innd /usr/bin/innernet-server {
  /usr/bin/innernet-server rm,
  /etc/innernet-server/ rw,
  /etc/innernet-server/* rw,
  /var/lib/innernet-server/ rw,
  /var/lib/innernet-server/* rwlk,

  # Runs within same profile to work with NoNewPrivileges=yes
  /usr/bin/ip ix,
  /usr/bin/ip rm,

  /usr/lib/** rm,
  /etc/ld.so.* r,
  /dev/urandom r,
  /dev/null rw,
  owner @{PROC}/@{pid}/{maps,cgroup} r,

  network inet,
  network inet6,
  network netlink raw,
  network unix, # systemd journal stdout/stderr

  capability net_admin,
  signal (receive) peer=unconfined,
}
```

It does not include rules for running userspace wireguard, as I don't have it here.
Profile for SELinux can probably work with kernel backend using roughly same rules, translated to its own syntax.

I think "network netlink raw," should still allow most CAP_NET_ADMIN abuse outlined above, but don't think AppArmor supports more granularity there either, still maybe whitelist path and executable restrictions there are better than a broad read-only fs access.

One way to make it a bit safer with this kind of profile granularity at the cost of increased complexity might be to split all netlink operations into a separate very simple binary which does not have code for doing anything outside of what innernet needs, and only run that with net_admin plus env scrubbing that AppArmor does in "Cx" rules.

(forgot to test-load profile with abi spec initially, and haven't rebooted for it to be loaded that way, so did repost with fixes after more testing)

**Comment by strohel** [comment]
I've tried non-root innernet with v1.4.0-beta.2 that includes #94 and it works well!

One small papercut is that tailing to `setcap` on the innernet binary no longer provides a nice error msg:
```bash
strohel@mat480s ~/work/innernet $ innernet up tonari
[*] bringing up the interface.

[E] Operation not permitted (os error 1)
```

**Comment by mcginty** [comment]
Closed in v1.4.0
